bugsnag-plugin-android-ndk/src/main/jni/event.c

```diff
@@ -142,6 +138,26 @@ static int allocate_metadata_index(bugsnag_metadata *metadata,
   return index;
 }
 
+void bsg_free_metadata_value(bsg_metadata_value *value) {
+  if (value->type == BSG_METADATA_OPAQUE_VALUE &&
+      value->opaque_value_size > 0) {
+    bsg_free(value->opaque_value);
+
+    value->opaque_value = NULL;
+    value->opaque_value_size = 0;
```

Comment:
Quick question: If an opaque value has been cleared, is its type not now `BSG_METADATA_NONE_VALUE`?

Reply:
That is left up to the caller, since it's not required for the functions that clear the entire metadata or all breadcrumbs.